source/pidomus.cc

```diff
@@ -997,24 +1003,53 @@ piDoMUS<dim, spacedim, LAC>::solve_jacobian_system(const double /*t*/,
     }
   else
     {
+      unsigned int tot_iteration = 0;
+      try
+        {
+          if (use_double_preconditioner && verbose)
+            pcout << " --> Coarse preconditioner "
+                  << std::endl;
+          PrimitiveVectorMemory<typename LAC::VectorType> mem;
+          SolverControl solver_control (matrices[0]->m(),
+                                        jacobian_solver_tolerance);
 
-      PrimitiveVectorMemory<typename LAC::VectorType> mem;
-      SolverControl solver_control (matrices[0]->m(),
-                                    jacobian_solver_tolerance);
+          SolverFGMRES<typename LAC::VectorType>
+          solver(solver_control, mem,
+                 typename SolverFGMRES<typename LAC::VectorType>::AdditionalData(50, true));
 
-      SolverFGMRES<typename LAC::VectorType>
-      solver(solver_control, mem,
-             typename SolverFGMRES<typename LAC::VectorType>::AdditionalData(50, true));
+          auto S_inv = inverse_operator(jacobian_op, solver, jacobian_preconditioner_op);
+          S_inv.vmult(dst, src);
 
-      auto S_inv = inverse_operator(jacobian_op, solver, jacobian_preconditioner_op);
-      S_inv.vmult(dst, src);
+          tot_iteration += solver_control.last_step();
+        }
+      catch (...)
```

Comment:
Così facendo aumenti solo il numero di iterazioni del solver. Se non converge con un numero di iterazioni uguali alla dimensione della matrice non dovrebbe convergere nemmeno con il doppio… a meno che non cambi precondizionatore


On 25 February 2016 at 23:12:44, Alberto Sartori (notifications@github.com) wrote:

In source/pidomus.cc:

>   
> -      auto S_inv = inverse_operator(jacobian_op, solver, jacobian_preconditioner_op);
> -      S_inv.vmult(dst, src);
> +          tot_iteration += solver_control.last_step();
> +        }
> +      catch (...)
qui dentro non faccio niente se non uso double_preconditioner. proporrei

if (use_double_preconditioner)
{
pcout finer preconditioner;
}
else
{
jacobian_preconditioner_finer = jacobian_preconditioner;
}

e il resto fuori dall'if

—
Reply to this email directly or view it on GitHub.